bindings/node/lib/bindings/tokenizer.test.ts

```diff
@@ -131,6 +132,7 @@ describe("Tokenizer", () => {
 
       const model = BPE.empty();
       tokenizer = new Tokenizer(model);
+      tokenizer.setPreTokenizer(whitespacePreTokenizer());
```

Comment:
I don't understand why you want to add these in all the tests. The pre-tokenizer won't get called anyway because all the tokens in the tests come are handled by the `AddedVocabulary`.

Reply:
That's not True, at least it was not in my case, all the tests were failing because the test string was not splitted through whitespace, so the added vocabulary was not handled, and those tests were failing because UNK_TOKEN was not defined.

We probably should really test the output of those methods in the full to make sure it's consistent, I'm pretty sure all those tests would have caught the missing unk tokens otherwise.

Reply:
The added vocabulary is always handled since it happens first, so no need for a pre-tokenizer. The tests actually started to fail because the BPE is receiving the whitespace between each word, and it doesn't have an `unk_token`. With the `Whitespace` pre-tokenizer, these whitespaces are just removed.

> We probably should really test the output of those methods in the full to make sure it's consistent, I'm pretty sure all those tests would have caught the missing unk tokens otherwise.

The unk token was treated this way by design. When provided we use it, otherwise the unknown tokens just get ignored. This is not a bug we didn't catch. That being said, feel free to add all the tests you find necessary, we never have enough of them :slightly_smiling_face: